### DISCUSSION TITLE
What removes things from the nodes array in interpeter.js?

### DISCUSSION THREAD

**Discussion by samlh** [DioxusLabs/dioxus#448] [Q&A]
Hi,

I've been reading through the source to try and understand how things work, and I had a question while reading through the interpreter code.

As far as I can tell, nothing ever removes things from the `this.nodes` array. I did consider that SetNode might be used for that, but it doesn't look like it. Is this correct, and if so, would it be beneficial to add a `this.nodes[root] = null` or similar in the `Remove` edit implementation to free up memory?

https://github.com/DioxusLabs/dioxus/blob/d9e14db434ee78465725d3ac24a455413f3fa963/packages/interpreter/src/interpreter.js#L55-L60

**Comment by jkelleyrtp** [comment]
Hey!

It would be beneficial to set the entry to null in Remove. .remove just removes the node from the DOM, but doesn't clear the entry.

In the VDOM, we use an intrusive arena that replaces dead slots as new elements are created. Old elements won't be freed until new elements are created that replace them in the `this.nodes` list. This means that there *is* garbage collection, but the this.nodes list will grow as large as the largest footprint of the DOM.

Additionally, `Remove` is not recursive, so there's no guarantee children will freed either.

I couldn't see setting `this.nodes[root] = null` being harmful though, so we might as well add it.
